BEPs/BEP-334.md

```diff
@@ -0,0 +1,100 @@
+<pre>
+  BEP: 334
+  Title: Greenfield CrossChain Permission Module
+  Status: Draft
+  Type: Standards
+  Created: 2023-12-06
+</pre>
+
+# BEP-334: Greenfield CrossChain Permission Module
+
+<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=6 orderedList=false} -->
+
+<!-- code_chunk_output -->
+
+- [BEP-334: Greenfield CrossChain Permission Module](#bep-334-greenfield-crosschain-permission-module)
+    - [1. Summary](#1-summary)
+    - [2. Abstract](#2-abstract)
+    - [3. Motivation](#3-motivation)
+    - [4. Specification](#4-specification)
+      - [4.1 Definitions](#41-definitions)
+      - [4.2 Interface](#42-interface)
+        - [4.2.1 Create Policy](#421-create-policy)
+        - [4.2.2 Delete Policy](#422-delete-policy)
+    - [5. License](#5-license)
+
+<!-- /code_chunk_output -->
+
+
+
+## 1. Summary
+Currently, Greenfield supports creating buckets and groups from the smart contracts deployed on BSC/opBNB. This means that the owner of a bucket can be a contract address. However, contract addresses cannot upload objects or put policies under the bucket, resulting in empty buckets owned by smart contracts.
+To address this issue, we are going to introduce the cross-chain permission module. With this BEP, smart contracts can grant EOA addresses the ability to upload objects under their buckets, and even support additional functions.
+
+## 2. Abstract
+Greenfield CrossChain Permission Module has several significant differences and improvements compared to current implementations:
+- BSC/opBNB account is allowed to put policy for Greenfield resources.
+- BSC/opBNB account is allowed to disable policy for Greenfield resources. 
+
+## 3. Motivation
+In the current framework, users could only change resource permission on Greenfield. The proposal allows Greenfield users to change resource permission directly on the BSC/opBNB network.
```

Comment:
It is really hard to understand what is `resource permission`.   As we already support grant read/write permission on BSC directly, what is the difference between it and the `resource permission` here?

Reply:
modified